### PR TITLE
Added mutex locks to protocol node and devices

### PR DESCRIPTION
**Pull request by stamp**
Please test this before merge.. not 100% sure it wont break stuff

### CODE REVIEW

File: nodes/stampzilla-chromecast/state.go [hunk 1]
```diff
@@ -70,8 +70,10 @@ func (s *State) Remove(c *Chromecast) {
 		s.Unlock()
 	}
 
-	if dev, ok := s.node.Devices_[c.Id]; ok {
+	if dev := s.node.Devices_.ByID(c.Id); dev != nil {
+		dev.Lock()
 		dev.Online = false
```
Comment (jonaz) [inline nodes/stampzilla-chromecast/state.go]: Perhaps add setters instead?

File: nodes/stampzilla-server/protocol/devices.go [hunk 2]
```diff
@@ -80,7 +80,7 @@ func (n *Devices) SetOfflineByNode(nodeUUID string) (list []*devices.Device) {
 	defer n.Unlock()
 
 	list = make([]*devices.Device, 0)
-	for _, dev := range n.devices {
+	for _, dev := range n.devices.All() {
 		if dev.Node == nodeUUID {
 			dev.Online = false
```
Comment (jonaz) [inline nodes/stampzilla-server/protocol/devices.go]: No lock here? Perhaps add SetOnline(bool) and lock there instead?

**Comment by jonaz** [comment]
@stamp i fixed the feedback i had in a new commit

**Comment by stamp** [comment]
Nice! :D

**Pull request merged**